test/unit/puppetlabs/puppetserver/lockable_pool_test.clj

```diff
@@ -185,9 +185,33 @@
         (is (thrown? IllegalStateException (.borrowItem pool)))
         ;; (let [instance (.borrowItem pool)]
         ;;   (is (true? true))
-        ;;   (.returnItem pool instance))
+        ;;   (.releaseItem pool instance))
 
         (is (true? true))
         (.unlock pool)
         @borrow-thread-1
         @borrow-thread-2))))
+
+(deftest pool-release-item-test
+  (testing (str "releaseItem call with value 'false' does not return item to "
+                "pool but does allow pool to still be lockable")
+    (let [pool (create-populated-pool 1)
+          instance (.borrowItem pool)]
+      (is (= 0 (.size pool)))
+      (.releaseItem pool instance false)
+      (is (= 0 (.size pool)))
+      (.lock pool)
+      (is (true? true))
+      (.unlock pool)
+      (is (true? true))))
+  (testing (str "releaseItem call with value 'true' does not return item to "
+                "pool and allows pool to still be lockable")
+    (let [pool (create-populated-pool 1)
+          instance (.borrowItem pool)]
+      (is (= 0 (.size pool)))
+      (.releaseItem pool instance true)
+      (is (= 1 (.size pool)))
+      (.lock pool)
+      (is (true? true))
```

Comment:
Would be worth a quick check to see if we could replace these with some sort of 'is lock held' assertion.  Ruth had a pattern for this in one of her tests but I think it might rely on having access to the full service context, so maybe not worth trying to duplicate here.